sql/src/main/java/io/crate/operation/aggregation/impl/SumAggregation.java

```diff
@@ -31,57 +31,103 @@
 import io.crate.types.DataTypes;
 import org.elasticsearch.common.breaker.CircuitBreakingException;
 
+import javax.annotation.Nullable;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiFunction;
 
-public class SumAggregation extends AggregationFunction<Double, Double> {
+public class SumAggregation<T extends Number> extends AggregationFunction<T, T> {
 
     public static final String NAME = "sum";
 
     private final FunctionInfo info;
+    private final BiFunction<T, T, T> addition;
+    private final DataType<T> type;
+    private final int bytesSize;
+
+    public static final Map<DataType, SumAggregation> sumFunctions;
+
+    static {
+        Map<DataType, SumAggregation> m = new HashMap<>();
```

Comment:
I've changed it in a way that all the SumAggregations are created during registering process.
I've also removed the map.
Since the implementations may be needed elsewhere, I would have several static instances in this class, rather than creating and also reimplementing this addition lambda elsewhere again. The constructor could remain private then, and the using program could only use the already existing instances.
@mfussenegger your thoughts?